packages/next/build/swc/index.js

```diff
@@ -0,0 +1,71 @@
+const { loadBinding } = require('@node-rs/helper')
+const path = require('path')
+
+/**
+ * __dirname means load native addon from current dir
+ * 'next-swc' is the name of native addon
+ * the second arguments was decided by `napi.name` field in `package.json`
+ * the third arguments was decided by `name` field in `package.json`
+ * `loadBinding` helper will load `next-swc.[PLATFORM].node` from `__dirname` first
+ * If failed to load addon, it will fallback to load from `next-swc-[PLATFORM]`
+ */
+const bindings = loadBinding(
```

Comment:
BTW. `@node-rs/helper` is facing a problem with `ncc`: https://github.com/napi-rs/node-rs/issues/316
I haven't come up with a solution to this problem

Reply:
Going to share this internally, thanks for sharing!

Reply:
Just had a look at this and I'd suggest treating `@node-rs/crc32` etc as external dependencies and directly making these dependencies of Next.js itself. That way the `optionalDependencies` install pattern which avoids including unnecessary binaries can be kept for Next.js installs. I've provided some more detail in https://github.com/napi-rs/node-rs/issues/316#issuecomment-887076534. There could be a solution to get these to build but it would require code changes to ensure analyzable wildcard requires and not using optionalDependencies.